app/src/androidTest/java/org/wikipedia/main/SmokeTests.kt

```diff
@@ -516,15 +796,401 @@ class SmokeTests {
         // Remove article from reading list
         onView(withText("Remove from Saved")).perform(click())
 
+        TestUtil.delay(5)
+
+        // Back to reading list screen
+        pressBack()
+
+        TestUtil.delay(2)
+
+        // Back to `Saved` tab
+        pressBack()
+
+        TestUtil.delay(2)
+
+        // Test history clearing feature - Go to search tab
+        onView(allOf(withId(R.id.nav_tab_search), withContentDescription("Search"),
+            childAtPosition(childAtPosition(withId(R.id.main_nav_tab_layout), 0), 2), isDisplayed())).perform(click())
+
+        TestUtil.delay(2)
+
+        // Click on `Clear history` icon
+        onView(allOf(withId(R.id.history_delete), withContentDescription("Clear history"), isDisplayed())).perform(click())
+
+        TestUtil.delay(2)
+
+        // Assert deletion message
+        onView(allOf(withId(androidx.appcompat.R.id.alertTitle), isDisplayed())).check(matches(withText("Clear browsing history")))
+
+        TestUtil.delay(2)
+
+        onView(allOf(withId(android.R.id.button2), withText("No"), isDisplayed())).perform(scrollTo(), click())
+
         TestUtil.delay(2)
 
+        // Turn off airplane mode
         TestUtil.setAirplaneMode(false)
 
+        TestUtil.delay(5)
+
+        // Click the More menu
+        onView(allOf(withId(R.id.nav_more_container), isDisplayed())).perform(click())
+
+        TestUtil.delay(1)
+
+        // Log-in the user
+        // Click the Login menu item
+        onView(allOf(withId(R.id.main_drawer_login_button), isDisplayed())).perform(click())
+
+        TestUtil.delay(2)
+
+        // Click the login button
+        onView(allOf(withId(R.id.create_account_login_button), withText("Log in"), isDisplayed()))
+            .perform(click())
+
+        TestUtil.delay(2)
+
+        // Type in an incorrect username and password
```

Comment:
It looks like all the following actions depend on a _correct_ username and password, which makes this comment confusing.